### ISSUE TITLE
Cannot use well-known attribute names with custom elements

### ISSUE THREAD

**Issue by hesxenon** [syntax-tree/hast-util-to-html#45]
### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=issues) and [discussions](https://github.com/orgs/syntax-tree/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

https://esm.sh/hast-util-to-html@9?bundle

### Steps to reproduce

See here: https://jsbin.com/jelewudote/3/edit?html,console,output

<details>
<summary><i>Copy Pasta'd repro for later reference</i></summary>

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Bin</title>
</head>
<body>
  <script>
    console.clear();
    class Foo extends HTMLElement {
      connectedCallback() {
        console.log(`rendered: ${this.getAttribute("selected")}`);
      }
    }
    customElements.define("my-foo", Foo)
  </script>

  <script type="module">
    import {toHtml} from "https://esm.sh/hast-util-to-html@9?bundle"

    console.log(`serialized: ${toHtml({
      type: "element",
      children: [],
      properties: {
        selected: "some value",
      },
      tagName: "my-foo",
    })}`)
  </script>
  
  <my-foo selected="some value"></my-foo>
</body>
</html>
```

</details>

Basically:
1. define a custom element which relies on the value of an attribute that has a well-known name
2. serialize such a custom element with an "unexpected" value for the well-known name

### Actual behavior

Serialization enforces the well-known definition over its intended usage.

> `"serialized: <my-foo selected></my-foo>"`

### Expected behavior

Serialization should not enforce the well-known definition in the case of custom elements. It should align with what is rendered by the browser.

> `"serialized: <my-foo selected="some value"></my-foo>"`

### Runtime

Firefox/Chrome

**Comment by wooorm** [comment]
Could you take a second to look at which exact package the problem is? Might not be this exact package. If it is this package, a broken test is nice.

Though, I'd think:
- Weird values for a Boolean should indeed be passed through as is
- Actually select=select, I can see that being turned into select, which I think custom elements should support?

**Comment by hesxenon** [comment]
As far as I tracked it (sorry, on the phone) the behaviour comes from
serializeAttribute in combination with the upstream property-information
library.

The problem is, the way it's written it's not exactly a fault per se - more
like a situation that wasn't thought of (and probably was impossible to
foresee at the time).

Property information delivers the correct information for the selected
attribute because it doesn't (as far as I can tell) have the tagname
available.

hast-util-to-html on the other hand could just skip querying
property-information for custom elements, the question is whether that is
the correct solution.

**Comment by hesxenon** [comment]
Sure, let me check that out in a bit more depth tomorrow

**Comment by wooorm** [comment]
Released in [`9.0.4`](https://github.com/syntax-tree/hast-util-to-html/releases/tag/9.0.4)!
